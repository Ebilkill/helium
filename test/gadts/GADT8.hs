data X a where
    I :: Int -> X Int
    B :: Bool -> X Bool
    XEQ :: a -> a -> X Bool

f :: X a -> X b -> Int 
f x y = case x of 
        I a -> case y of 
            I b -> a + b
            B b -> a + b