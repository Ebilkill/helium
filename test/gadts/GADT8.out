Prelude is up to date
Compiling gadts/GADT8.hs
fromList []
Could not unify 'Bool ~ 'Int (different head constructors)


(10,13): Type error in pattern application
 pattern          : B b
 constructor      : B
   type           : Bool -> X Bool
   does not match : Int  -> X Int 

Compilation failed with 1 error
